drivers/storage/portworx/status.go

```diff
@@ -433,3 +450,70 @@ func isTLSEnabled() bool {
 	enabled, err := strconv.ParseBool(os.Getenv(envKeyPortworxEnableTLS))
 	return err == nil && enabled
 }
+
+func (p *portworx) getOperatorToken(secretkey string) (string, error) {
+	claims := &auth.Claims{
+		Issuer:  "openstorage.io",
+		Subject: "operator@openstorage.io",
+		Name:    "operator communications",
+		Email:   "operator@openstorage.io",
+		Roles:   []string{"system.admin"},
+		Groups:  []string{"*"},
+	}
+
+	signature, err := auth.NewSignatureSharedSecret(string(secretkey))
+	if err != nil {
+		return "", err
+	}
+	token, err := auth.Token(claims, signature, &auth.Options{
+		Expiration: time.Now().Add(1 * time.Hour).Unix(),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+func (p *portworx) getAuthSecret() (string, error) {
+	// Get secret for operator system shared secret
+	pxSharedSecretName := os.Getenv(envKeyPortworxAuthSecretName)
```

Comment:
The user will not pass these variables to the operator. We will have to read it from the StorageCluster's env.

Reply:
Fixed this